bigchaindb/core.py

```diff
@@ -516,94 +408,16 @@ def write_block(self, block):
-        transaction = Transaction.create([self.me], [([self.me], 1)],
-                                         metadata=metadata)
-
-        # NOTE: The transaction model doesn't expose an API to generate a
-        #       GENESIS transaction, as this is literally the only usage.
-        transaction.operation = 'GENESIS'
-        transaction = transaction.sign([self.me_private])
-
-        # create the block
-        return self.create_block([transaction])
-
-    def create_genesis_block(self):
-        """Create the genesis block
-
-        Block created when bigchain is first initialized. This method is not atomic, there might be concurrency
-        problems if multiple instances try to write the genesis block when the BigchainDB Federation is started,
-        but it's a highly unlikely scenario.
-        """
-
-        # 1. create one transaction
-        # 2. create the block with one transaction
-        # 3. write the block to the bigchain
-
-        blocks_count = backend.query.count_blocks(self.connection)
-
-        if blocks_count:
-            raise exceptions.GenesisBlockAlreadyExistsError('Cannot create the Genesis block')
-
-        block = self.prepare_genesis_block()
-        self.write_block(block)
-
-        return block
-
-    def vote(self, block_id, previous_block_id, decision, invalid_reason=None):
-        """Create a signed vote for a block given the
-        :attr:`previous_block_id` and the :attr:`decision` (valid/invalid).
-
-        Args:
-            block_id (str): The id of the block to vote on.
-            previous_block_id (str): The id of the previous block.
-            decision (bool): Whether the block is valid or invalid.
-            invalid_reason (Optional[str]): Reason the block is invalid
-        """
-
-        if block_id == previous_block_id:
-            raise exceptions.CyclicBlockchainError()
-
-        vote = {
-            'voting_for_block': block_id,
-            'previous_block': previous_block_id,
-            'is_block_valid': decision,
-            'invalid_reason': invalid_reason,
-            'timestamp': gen_timestamp()
-        }
-
-        vote_data = serialize(vote)
-        signature = crypto.PrivateKey(self.me_private).sign(vote_data.encode())
-
-        vote_signed = {
-            'node_pubkey': self.me,
-            'signature': signature.decode(),
-            'vote': vote
-        }
-
-        return vote_signed
-
     def write_vote(self, vote):
         """Write the vote to the database."""
         return backend.query.write_vote(self.connection, vote)
 
-    def get_last_voted_block(self):
-        """Returns the last block that this node voted on."""
-
-        last_block_id = backend.query.get_last_voted_block_id(self.connection,
-                                                              self.me)
-        return Block.from_dict(self.get_block(last_block_id))
-
     def block_election(self, block):
```

Comment:
May be remove `block_election` too?

Reply:
That is not the context of this PR.

Reply:
Yes, Will address in a separate PR.

Reply:
Actually, I removed this in my latest change. Will address anymore code removal in upcoming PRs.